### PR TITLE
switching up the way mute/unmute works

### PR DESCRIPTION
**Pull request by benlower**
removing mute(), unmute(), and isMuted(). adding muteMic(), unmuteMic(), isMicMuted(), muteSpeaker(), unmuteSpeaker(), and isSpeakerMuted().

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -172,85 +178,44 @@ export class UltravoxSession {
...
-
-        default:
-          break;
-      }
+  muteMic(): void {
+    if (!this.room?.localParticipant) {
+      throw new Error('Cannot muteMic.');
     }
+    this._isMicMuted = true;
+    this.room.localParticipant.setMicrophoneEnabled(!this._isMicMuted);
```
Comment (mdepinet) [inline src/index.ts:186]:
```suggestion
    this.room.localParticipant.setMicrophoneEnabled(false);
```

(just seems more clear)

File: src/index.ts [hunk 2]
```diff
@@ -172,85 +178,44 @@ export class UltravoxSession {
...
-    for (const role of roleSet) {
-      if (!Object.values(Role).includes(role)) {
-        throw new Error(`Invalid role: ${role}`);
-      }
+  unmuteMic(): void {
+    if (!this.room?.localParticipant) {
+      throw new Error('Cannot unmuteMic.');
+    }
+    this._isMicMuted = false;
+    this.room.localParticipant.setMicrophoneEnabled(!this._isMicMuted);
```
Comment (mdepinet) [inline src/index.ts:194]:
```suggestion
    this.room.localParticipant.setMicrophoneEnabled(true);
```

File: src/index.ts [hunk 3]
```diff
@@ -172,85 +178,44 @@ export class UltravoxSession {
...
+  muteSpeaker(): void {
+    if (!this.room?.remoteParticipants) {
+      throw new Error('Cannot muteSpeaker.');
     }
+    this._isSpeakerMuted = true;
+    this.room.remoteParticipants.forEach((rp) => {
+      rp.audioTrackPublications.forEach((atp) => {
+        atp.track?.setMuted(this._isSpeakerMuted);
+      });
+    });
```
Comment (mdepinet) [inline src/index.ts:206]:
```suggestion
    this.room.remoteParticipants.forEach((participant) => {
      participant.audioTrackPublications.forEach((publication) => {
        publication.track?.setMuted(true);
      });
    });
```

nit: prefer words to acronyms for variable names for better readability

File: src/index.ts [hunk 4]
```diff
@@ -172,85 +178,44 @@ export class UltravoxSession {
...
-          break;
-      }
+  muteSpeaker(): void {
+    if (!this.room?.remoteParticipants) {
+      throw new Error('Cannot muteSpeaker.');
     }
+    this._isSpeakerMuted = true;
+    this.room.remoteParticipants.forEach((rp) => {
+      rp.audioTrackPublications.forEach((atp) => {
+        atp.track?.setMuted(this._isSpeakerMuted);
```
Comment (mdepinet) [inline src/index.ts:204]: Double checking: In LiveKit's other SDKs I had to setDisabled as "muted" is a server-controlled property.  Have you been able to test this?

File: src/index.ts [hunk 5]
```diff
@@ -172,85 +178,44 @@ export class UltravoxSession {
...
+  unmuteSpeaker(): void {
+    if (!this.room?.remoteParticipants) {
+      throw new Error('Cannot unmuteSpeaker.');
     }
+    this._isSpeakerMuted = false;
+    this.room.remoteParticipants.forEach((rp) => {
+      rp.audioTrackPublications.forEach((atp) => {
+        atp.track?.setMuted(this._isSpeakerMuted);
+      });
+    });
```
Comment (mdepinet) [inline src/index.ts:218]:
```suggestion
    this.room.remoteParticipants.forEach((participant) => {
      participant.audioTrackPublications.forEach((publication) => {
        publication.track?.setMuted(false);
      });
    });
```

**Review by mdepinet — APPROVED** [review]

File: src/index.ts (hunk 4 above)
Comment (benlower) [reply src/index.ts]: yes i tested in our main demo app. added a "mute speaker" button and can toggle on/off as expected.

**Pull request merged**